cedar-policy/src/frontend/is_authorized.rs

```diff
@@ -212,6 +308,45 @@ impl AuthorizationCall {
         let (policies, entities) = self.slice.try_into(schema.as_ref())?;
         Ok((q, policies, entities))
     }
+
+    #[cfg(feature = "partial-eval")]
+    fn get_components_partial(self) -> Result<(Request, PolicySet, Entities), Vec<String>> {
+        let schema = self
+            .schema
+            .map(|v| Schema::from_json_value(v.into()))
+            .transpose()
+            .map_err(|e| [e.to_string()])?;
+        let principal = match self.principal {
+            Some(p) => Some(
+                EntityUid::from_json(p.into())
+                    .map_err(|e| ["Failed to parse principal".into(), e.to_string()])?,
+            ),
+            None => None,
+        };
+        let action = EntityUid::from_json(self.action.into())
+            .map_err(|e| ["Failed to parse action".into(), e.to_string()])?;
+        let resource = match self.resource {
+            Some(r) => Some(
+                EntityUid::from_json(r.into())
+                    .map_err(|e| ["Failed to parse resource".into(), e.to_string()])?,
+            ),
+            None => None,
+        };
+
+        let context = serde_json::to_value(self.context)
+            .map_err(|e| [format!("Error encoding the context as JSON: {e}")])?;
+        let context = Context::from_json_value(context, schema.as_ref().map(|s| (s, &action)))
+            .map_err(|e| [e.to_string()])?;
```

Comment:
The this logic be easily shared with `get_components`?

Reply:
Not with my level of proficiency in Rust. I think something like this would work with similar code for get_component_partial(). But I'm getting move issues and I'm not sure to get over them.
```
impl AuthorizationCall {
    fn parse_schema(self) -> Result<Option<Schema>, Vec<String>> {
        self.schema
            .map(|v| Schema::from_json_value(v.into()))
            .transpose()
            .map_err(|e| vec!(e.to_string()))
    }
    ...
    fn get_components(self) -> Result<(Request, PolicySet, Entities), Vec<String>> {
        let schema = self.parse_schema()?;
        let principal = self.parse_principal()?;
        let action = self.parse_action()?;
        let resource = self.parse_resource()?;
        let context = self.parse_context(schema, &action)?;
        ...
    }
}
```

Reply:
@john-h-kastner-aws I pushed another commit that does some factoring out. Let me know what you think.